provider/aci_test.go

```diff
@@ -1191,3 +1191,151 @@ func TestCreatePodWithProjectedVolume(t *testing.T) {
 		t.Fatal("Failed to create pod", err)
 	}
 }
+
+func TestCreatePodWithCSIVolume(t *testing.T) {
+	podName := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+
+	aadServerMocker := NewAADMock()
+	aciServerMocker := NewACIMock()
+
+	aciServerMocker.OnGetRPManifest = func() (int, interface{}) {
+		manifest := &aci.ResourceProviderManifest{
+			Metadata: &aci.ResourceProviderMetadata{
+				GPURegionalSKUs: []*aci.GPURegionalSKU{
+					{
+						Location: fakeRegion,
+						SKUs:     []aci.GPUSKU{aci.K80, aci.P100, aci.V100},
+					},
+				},
+			},
+		}
+
+		return http.StatusOK, manifest
+	}
+
+	mockCtrl := gomock.NewController(GinkgoT())
+	podLister := NewMockPodLister(mockCtrl)
+	secretLister := NewMockSecretLister(mockCtrl)
+	secretNamespaceLister := NewMockSecretNamespaceLister(mockCtrl)
+	configMapLister := NewMockConfigMapLister(mockCtrl)
+	serviceLister := NewMockServiceLister(mockCtrl)
+	pvcLister := NewMockPersistentVolumeClaimLister(mockCtrl)
+	pvLister := NewMockPersistentVolumeLister(mockCtrl)
+
```

Comment:
why are all the variables above called mockxxx while those are not called mock?

Reply:
I followed the code convention. I'll go ahead and follow the right convention for this case.